CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -382,15 +364,8 @@ public function submit($submittedValues) {
     }
 
     $statusMsg = ts('The payment record has been processed.');
-    // send email
-    if (!empty($result) && !empty($this->_params['is_email_receipt'])) {
-      $this->_params['contact_id'] = $this->_contactId;
-      $this->_params['contribution_id'] = $this->_contributionId;
-
-      $sendReceipt = $this->emailReceipt($this->_params);
-      if ($sendReceipt) {
-        $statusMsg .= ' ' . ts('A receipt has been emailed to the contributor.');
-      }
+    if (empty($payment['is_error']) && !empty($this->_params['is_email_receipt'])) {
```

Comment:
So I think the payment params could be modified by hooks once Payment.create has been called but we are relying on the value of `is_email_receipt` from before that function is called.  Are we able to use the returned parameters of Payment.create to check if `is_email_receipt` stayed TRUE and therefore an email actually got sent?

Reply:
ok, then I'm ok with this.